Switched react-katex package

CFrez opened the pull request:
This resolves some issues with the rendering of latex. #2086 and  #2556 are both resolved with this fix.

The existing [react-katex](https://github.com/talyssonoc/react-katex) package seems to have been abandoned 3 years ago. This switches to a newer [react-katex](https://github.com/MatejBransky/react-katex) package that was inspired by the first. The new package is what is currently recommended on the [react-markdown](https://github.com/remarkjs/react-markdown#use-a-plugin-and-custom-renderers-math) page. [This](https://github.com/talyssonoc/react-katex/issues/49) shows the difference between the two packages.

I manually tested the functionality of each issue:

#2086 
<img width="645" alt="2086" src="https://user-images.githubusercontent.com/30352852/105205423-df5ac300-5b0a-11eb-8f2b-03b432a503f3.png">

#2556 
<img width="653" alt="2556" src="https://user-images.githubusercontent.com/30352852/105205444-e5e93a80-5b0a-11eb-82e7-a41d96ee0e28.png">

Both pytest and jstest ran without problems. 

While running the e2e test I did receive this failure:
![disable widgets -- disconnects the client and disables widgets (failed)](https://user-images.githubusercontent.com/30352852/105206597-2ac1a100-5b0c-11eb-8efc-2a227550f53a.png)

I ran the e2e on the develop branch to compare and it seems to fail at different places. I am not sure if this is something with my setup or an ongoing issue. I am fairly confident that the issue is not within the changes proposed by this pull request.

Please let me know if there is a better way to test these changes and I can provide the update.

karriebear committed:
Fix slider UX (#2596)

* Fix slider UX

* Add snapshot

tvst committed:
Revert "Deploy button front (#2552)" (#2599)

* Revert "Deploy button front (#2552)"

This reverts commit 0f66b6eea6083413e486ebdd414af3ba9e05ba4c.

* Fix merge (erm... revert) conflict.

* Revert "Deploy button py (#2535)"

This reverts commit 0bbc7aa681a15458708de31ab7569f2b16f244f9.

Co-authored-by: karrie <karrie@streamlit.io>

Commit:
Remove extra period from sentence (#2601)

vdonato committed:
Have st.number_input warn user of mismatched type and format string (#2604)

* Add %u to list of valid number_input formatters

* Have number_input warn user of mismatched type and format string

* Use more modern f-string syntax instead of .format()

CFrez committed:
switched react-katex package

AnOctopus committed:
Feature/sudoless metrics (#2605)

* Add and use sudoless machine id getter for metrics

Step 2 of the metrics user ID improvement plan.

* Add machine-id v3 tests

Duplicates of the v1 tests, since those covered the cases that remain in v3,
and we will remove v1 and its tests later.

Also fix formatting of some python files.

* DRY out machine ID paths into global constants

CFrez committed:
lockfile updated

akrolsmir committed:
Increase side padding to 5rem when app is in wide mode (#2613)

* Increase side padding to 5rem when app is in wide mode

* Upload Cypress snapshot

akrolsmir reviewed (approved):
Hi @RedFrez , thanks for your contribution! Really appreciated your explanations of why we're switching to this package and what we're gaining.

It does look like the st_markdown.spec.js needs a minor tweak (I'm amazed at that old misspelling, actually).

CFrez committed:
update test to corrected spelling

CFrez commented:
I was just tracking this down. it seems like whatever generated the MathML version spelled it wrong, and the test just matched it. 



<img width="1248" alt="Screen Shot 2021-01-20 at 4 30 50 PM" src="https://user-images.githubusercontent.com/30352852/105251741-b0ac0f00-5b41-11eb-88b8-668859fadda0.png">

LaTeX and TeX also had the same misspellings. 

![Screen Shot 2021-01-20 at 5 00 29 PM](https://user-images.githubusercontent.com/30352852/105251830-d802dc00-5b41-11eb-915b-14c8ec3da423.png)

All tests should pass now. The Block math didn't have this issues since it just had math.

vdonato committed:
Rerender Maybe components when they're first disabled (#2617)

\#2395 ended up being caused by this since we don't rerender placeholder
components created by `st.empty()` when react props/state change as a
performance optimization, but we need to do so when replacing an
existing component with an empty one to get rid of what's currently
drawn on screen.

Closes #2395

CFrez committed:
decapitalize KaTex

CFrez commented:
@akrolsmir the current test that is failing is with st.file_uploader timed out. Is this a known flaky test? I am unable to rerun the test as the option is disabled for me.

karriebear commented:
I've triggered a rerun of the test but it should be fixed in the latest `develop`. If it fails again, could you try pulling latest in to see if it helps?

akrolsmir reviewed (changes requested):
Hm, it looks like the latest merge has pulled in some unwanted files and the diff has ballooned as a result. Maybe you want to reset the git branch to before the merge, then pull in the latest develop? Something like:

```
git checkout latex
git reset --hard d3513118887a28a6765fe3998814a35bd08d009d
git merge upstream/develop
git push
```

CFrez committed:
Merge remote-tracking branch 'upstream/develop' into latex

CFrez commented:
@akrolsmir I flipped a coin between merge and rebase and I guess the wind blew it the wrong way.

Is there somewhere other than the main contributing page that has instructions about creating a pull request?

akrolsmir commented:
> @akrolsmir I flipped a coin between merge and rebase and I guess the wind blew it the wrong way.
> 
> Is there somewhere other than the main contributing page that has instructions about creating a pull request?

Hm, I don't believe so, but we definitely should have one -- thanks for the suggestion!

Anyways, this PR looks great; that one test failure is another flake (it's been fixed in the latest develop, I think) so I'm just rerunning for now :P

kmcgrady reviewed (approved):
Looks Great! Going to merge this in!
